### PR TITLE
Add sort button to cleanup page

### PR DESCRIPTION
**Pull request by colasnaudi**

### CODE REVIEW

File: bemserver_ui/templates/pages/services/cleanup/manage.html [hunk 1]
```diff
@@ -16,10 +16,10 @@ <h5>Campaign</h5>
         <div class="col">
             <h5>Service state</h5>
             {% if signed_user.is_admin %}
-            <input type="radio" class="btn-check" name="svc_state" id="svc_state_on" autocomplete="off"{% if cleanup_campaign.is_enabled %} checked{% endif %}>
-            <label class="btn btn-outline-success" for="svc_state_on">ON</label>
-            <input type="radio" class="btn-check" name="svc_state" id="svc_state_off" autocomplete="off"{% if not cleanup_campaign.is_enabled %} checked{% endif %}>
-            <label class="btn btn-outline-danger" for="svc_state_off">OFF</label>
+            <input type="radio" class="btn-check" name="svc_state" id="svc-state-on" autocomplete="off"{% if cleanup_campaign.is_enabled %} checked{% endif %}>
```
Comment (davidfrederique) [inline bemserver_ui/templates/pages/services/cleanup/manage.html:19]: No need to rename those ids.

**Comment by davidfrederique** [comment]
When pushing or submitting a pull request, we configured GitHub to execute some actions like lint, run tests or build/publish a new release. Here, in UI, there are no automatic tests on app features, just lint on code.
In this pull request lint check failed.
To avoid (or at least reduce) this, we use [pre-commit](https://pre-commit.com/), which runs some checks before local committing.
You should (must) install it in your Python virtual env. This will verify (and correct) the Python code syntax before even send it.
We only merge pull requests that passes, at least, GitHub action checks.

File: bemserver_ui/templates/pages/services/cleanup/manage.html [hunk 2]
```diff
@@ -36,18 +36,36 @@ <h5>Service state</h5>
         <div class="col">
             <div class="table-responsive-xl">
                 <table class="table table-sm table-hover table-bordered caption-top">
-                    <caption class="text-end">
-                        <small class="text-nowrap text-muted"><app-items-count first-item="1" last-item="{{ cleanup_timeseries | length }}" total-count="{{ cleanup_timeseries | length }}"></app-items-count></small>
-                    </caption>
+                    <div class="row mb-3">
```
Comment (davidfrederique) [inline bemserver_ui/templates/pages/services/cleanup/manage.html:39]: `<caption>` HTML tag, including items count, was fine.

File: bemserver_ui/templates/pages/services/cleanup/manage.html [hunk 3]
```diff
@@ -36,18 +36,36 @@ <h5>Service state</h5>
...
+                        <div class="col">
```
Comment (davidfrederique) [inline bemserver_ui/templates/pages/services/cleanup/manage.html:40]: Weird `<div class="col">` position. It is not supposed to be here...
This div should be inside `<caption>`, to display sort field on the left and items count on the right.

File: bemserver_ui/static/scripts/modules/views/services/cleanupManage.js [hunk 4]
```diff
@@ -23,15 +28,19 @@ class ServiceCleanuManageView {
...
         this.#initEventListeners();
     }
-
+    
     #cacheDOM() {
         this.#messagesElmt = document.getElementById("messages");
 
-        this.#stateOnRadioElmt = document.getElementById("svc_state_on");
-        this.#stateOffRadioElmt = document.getElementById("svc_state_off");
+        this.#stateOnRadioElmt = document.getElementById("svc-state-on");
```
Comment (davidfrederique) [inline bemserver_ui/static/scripts/modules/views/services/cleanupManage.js:35]: Those HTML elements ids do not need to be changed. So this update is not needed.

File: bemserver_ui/static/scripts/modules/views/services/cleanupManage.js [hunk 5]
```diff
@@ -46,6 +55,18 @@ class ServiceCleanuManageView {
...
+        for (let sortRadioElmt of this.#sortRadioElmts) {
+            sortRadioElmt.addEventListener("change", (event) => {
+                event.preventDefault();
+                let sortData = sortRadioElmt.id.split("-");
+                let newSortValue = `${sortData[2].toLowerCase() == "asc" ? "+" : "-"}${sortData[1].toLowerCase()}`;
+                if (this.#sortInputElmt.value != newSortValue) {
+                    this.#sortInputElmt.value = newSortValue;
+                    this.#formFiltersElmt.submit();
+                }
+                console.log("Sort Value : ",newSortValue);
```
Comment (davidfrederique) [inline bemserver_ui/static/scripts/modules/views/services/cleanupManage.js:67]: `console.log` or prints, used for debug purposes, must be remove before submitting code.

File: bemserver_ui/static/scripts/modules/views/services/cleanupManage.js [hunk 6]
```diff
@@ -93,11 +114,14 @@ class ServiceCleanuManageView {
             );
         }
     }
+
```
Comment (davidfrederique) [inline bemserver_ui/static/scripts/modules/views/services/cleanupManage.js:117]: Maybe this `refresh()` method is useless.

File: bemserver_ui/views/services/cleanup.py [hunk 7]
```diff
@@ -109,43 +109,61 @@ def _get_cleanup_data_for_campaign(campaign_id):
     )
 
 
-def _manage(cleanup_campaign, cleanup_campaign_etag):
+def _manage(cleanup_campaign, cleanup_campaign_etag, sort):
+    sort = sort
```
Comment (davidfrederique) [inline bemserver_ui/views/services/cleanup.py:113]: Useless line.

**Comment by lafrech** [comment]
Icing on the cake, you may want to either `rebase -i` or `git commit --amend` when everything is settled, for a cleaner history.

No need to keep the intermediary commits.

File: bemserver_ui/templates/pages/services/cleanup/manage.html (hunk 3 above)
Comment (davidfrederique) [reply bemserver_ui/templates/pages/services/cleanup/manage.html:40]: Should look something like this:

`<caption class="d-flex justify-content-between align-items-center">
    <form id="formFiltersElmt" class="d-lg-flex d-grid align-items-center gap-2" action="{{ url_for(request.endpoint) }}" method="POST">
        ...
    </form>
    <small class="text-nowrap text-muted">
        <app-items-count first-item="1" last-item="{{ cleanup_timeseries | length }}" total-count="{{ cleanup_timeseries | length }}"></app-items-count>
    </small>
</caption>
<thead>
    ...`

**Comment by colasnaudi** [comment]
Ok I'm going to change all these mistakes. I didn't knew about the `git commit --amend` command, i'll use it now.

File: bemserver_ui/templates/pages/services/cleanup/manage.html [hunk 8]
```diff
@@ -33,21 +33,37 @@ <h5>Service state</h5>
         </div>
     </div>
     <div class="row">
-        <div class="col">
+        <div class="col"> 
+            <div class="d-flex justify-content-between align-items-center">
+                <div class="d-flex align-items-center gap-2">
```
Comment (davidfrederique) [inline bemserver_ui/templates/pages/services/cleanup/manage.html:38]: This div is useless. `d-flex` class is already defined in form.

File: bemserver_ui/templates/pages/services/cleanup/manage.html [hunk 9]
```diff
@@ -33,21 +33,37 @@ <h5>Service state</h5>
...
+                        <label class="btn btn-sm btn-outline-info" for="sort-last_timestamp-asc" title="Timestamp date order"><i class="bi bi-sort-numeric-down"></i> Timestamp</label>
+                        <input type="radio" class="btn-check" name="sort" id="sort-last_timestamp-desc" autocomplete="off"{% if sort == "-last_timestamp" %} checked{% endif %}>
+                        <label class="btn btn-sm btn-outline-info" for="sort-last_timestamp-desc" title="Timestamp reverse date order"><i class="bi bi-sort-numeric-up"></i> Timestamp</label>
+                    </form>
+                </div>
+                <small class="text-nowrap text-muted">
+                    <app-items-count first-item="1" last-item="{{ cleanup_timeseries | length }}" total-count="{{ cleanup_timeseries | length }}"></app-items-count>
+                </small>
+            </div>
             <div class="table-responsive-xl">
```
Comment (davidfrederique) [inline bemserver_ui/templates/pages/services/cleanup/manage.html:56]: Close here the previous div row and open a new one.
Add a margin bottom (mb-2 for example) on the row before the table, to have a little space between sort buttons and table.

File: bemserver_ui/templates/pages/services/cleanup/manage.html [hunk 10]
```diff
@@ -33,21 +33,37 @@ <h5>Service state</h5>
         </div>
     </div>
     <div class="row">
```
Comment (davidfrederique) [inline bemserver_ui/templates/pages/services/cleanup/manage.html:35]: add class mb-2 on div row

File: bemserver_ui/views/services/cleanup.py [hunk 11]
```diff
@@ -109,43 +109,59 @@ def _get_cleanup_data_for_campaign(campaign_id):
...
 
 
-@blp.route("/campaign/<int:id>/state")
+@blp.route("/campaign/<int:id>/state", methods=["GET", "POST"])
 @auth.signin_required
 def campaign_state(id):
+    sort = "+timeseries_name"
+
+    if flask.request.method == "POST":
+        sort = flask.request.form["sort"]  #
```
Comment (davidfrederique) [inline bemserver_ui/views/services/cleanup.py:161]: useless #

**Pull request merged**